diff --git a/fwsrc/commonservices.c b/fwsrc/commonservices.c
index c6355c5..10ce40d 100644
--- a/fwsrc/commonservices.c
+++ b/fwsrc/commonservices.c
@@ -264,7 +264,7 @@ CMD_RET_TYPE cmd_Browse(char * buffer, char *pusrdata, unsigned short len, char
 #endif
 } // END: cmd_Browse(...)
 
-
+#if 0
 CMD_RET_TYPE cmd_GPIO(char * buffer, char *pusrdata, char * buffend)
 {
 	static const uint32_t AFMapper[16] = {
@@ -318,6 +318,7 @@ CMD_RET_TYPE cmd_GPIO(char * buffer, char *pusrdata, char * buffend)
 	}
 	return buffend - buffer;
 } // END: cmd_GPIO(...)
+#endif
 
 
 CMD_RET_TYPE cmd_Echo(char * pursdata, int retsize, unsigned short len, char * buffend)
@@ -715,10 +716,12 @@ int ICACHE_FLASH_ATTR issue_command(char * buffer, int retsize, char *pusrdata,
 		// Respond with device info or other general system things
 		case 'i': case 'I':
 		 	return cmd_Info( buffer, retsize, pusrdata, buffend );
+		case '7':
+			return cmd_SevenSeg( buffer, retsize, pusrdata, buffend );
 
 		// Set GPIO pin state
-		case 'g': case 'G':
-			return cmd_GPIO( buffer, pusrdata, buffend );
+		//case 'g': case 'G':
+		//	return cmd_GPIO( buffer, pusrdata, buffend );
 
 		// Echo command. E[data], responds with the same data.
 		case 'e': case 'E':
diff --git a/web/Makefile b/web/Makefile
index 0c08822..d0b5317 100644
--- a/web/Makefile
+++ b/web/Makefile
@@ -2,12 +2,13 @@ include ../user.cfg
 include ../esp82xx/common.mf
 
 .PHONY : all clean push
-all : execute_reflash push
+all : execute_reflash mfsmaker pushtodev
 
 CC = gcc
 CFLAGS = $(OPTS)
 LDLIBS = -lusb-1.0
-PAGE_SCRIPT = $(foreach s, $(call uniq, $(notdir $(wildcard page/jquery*gz) menuinterface.js $(PAGE_SCRIPTS))), <script language="javascript" type="text/javascript" src="$(s)"></script>)
+PAGE_SCRIPT = $(foreach s, $(call uniq, $(notdir menuinterface.js $(PAGE_SCRIPTS))),<script language="javascript" type="text/javascript" src="$(s)"></script>%%+n%%)
+#PAGE_SCRIPT = $(foreach s, $(call uniq, $(notdir $(wildcard page/jquery*gz) menuinterface.js $(PAGE_SCRIPTS))),<script language="javascript" type="text/javascript" src="$(s)"></script>%%+n%%)
 
 mfsmaker : mfsmaker.c
 pushtodev : pushtodev.c
@@ -22,8 +23,9 @@ page.mpfs : mfsmaker $(wildcard page/*) Makefile
 	# Inception level over 9000 for poor man's template substitution
 	$(info bla$ ${\n} blubb)
 	$(foreach p, $(filter PAGE_%,$(.VARIABLES)) $(filter PROJECT_%,$(.VARIABLES)) VERSSTR, \
-		sed -i "s/{{$(p)}}/$$(echo -n '$($(p))' | sed -e 's/[\/&"]/\\&/g')/" tmp/index.html && \
+		sed -i "s/{{$(p)}}/$$(/bin/echo -n '$($(p))' | sed -e 's/[\/&"]/\\&/g')/" tmp/index.html && \
 	) true
+	sed -i 's/%%+n%%/\n/g' tmp/index.html
 	./mfsmaker tmp page.mpfs
 
 push : pushtodev page.mpfs
diff --git a/web/page/menuinterface.js b/web/page/menuinterface.js
index ea3aea0..f300466 100644
--- a/web/page/menuinterface.js
+++ b/web/page/menuinterface.js
@@ -1,7 +1,10 @@
 //Copyright (C) 2015 <>< Charles Lohr, see LICENSE file for more info.
 //
 //This particular file may be licensed under the MIT/x11, New BSD or ColorChord Licenses.
+
 var wsUri = "ws://" + location.host + "/d/ws/issue";
+//var wsUri = "ws://" + "192.168.178.60" + "/d/ws/issue";
+
 var output;
 var websocket;
 var commsup = 0;
@@ -51,17 +54,13 @@ function init()
 	did_init = true;
 	GPIOlines = '';
 	for(var i=0; i<16; ++i)
-		GPIOlines += "<td align=center>"+ i
-			+ "<input type=button id=ButtonGPIO"+ i +" value=0 onclick=\"TwiddleGPIO("+ i +");\">"
-			+ "<input type=button id=BGPIOIn"+ i +" value=In onclick=\"GPIOInput("+ i +");\" class=\"inbutton\">"
-			+ "</td>";
 
 	$('#MainMenu > tbody:first-child').before( "\
 		<tr><td width=1> \
 		<input type=submit onclick=\"ShowHideEvent( 'SystemStatus' ); SystemInfoTick();\" value='System Status' id=SystemStatusClicker></td><td> \
 		<div id=SystemStatus class='collapsible'> \
 		<table width=100% border=1><tr><td> \
-<div id=output></div><div id=systemsettings></div> \n		</td></tr></table></div></td></tr>"
+		<div id=output></div><div id=systemsettings></div> \n		</td></tr></table></div></td></tr>"
 	);
 
 	$('#MainMenu > tbody:last-child').after( "\
@@ -92,13 +91,6 @@ function init()
 		<textarea id=custom_command_response readonly rows=15 cols=80></textarea> \
 		</td></tr></table></td></tr> \
 		 \
-		<tr><td width=1> \
-		<input type=submit onclick=\"ShowHideEvent( 'GPIOs' ); GPIODataTicker();\" value=\"GPIOs\"></td><td> \
-		<div id=GPIOs class=\"collapsible\"> \
-		<table width=100% border=1><tr>" +
- 		GPIOlines
-		+ "</tr></table></div></td></tr>\
-\
 		<tr><td width=1>\
 		<input type=submit onclick=\"ShowHideEvent( 'SystemReflash' );\" value=\"System Reflash\"></td><td>\
 		<div id=SystemReflash class=\"collapsible\">\
@@ -106,6 +98,11 @@ function init()
 		<input id=\"dragndropersystem\" type=\"file\" multiple> <div id=innersystemflashtext>Drop or browse for system (0x000.. 0x400...) or web (.mpfs) reflash files.</div>\
 		</div></div></td></tr>"
 	);
+	$("#custom_command").keyup(function(event) {
+		if (event.keyCode === 13) {
+			IssueCustomCommand();
+		}
+	});
 
 	MakeDragDrop( "InnerSystemReflash", DragDropSystemFiles );
 	$("#dragndropersystem").change(function() { DragDropSystemFiles(this.files ); });
@@ -186,6 +183,7 @@ function Ticker()
 			if( commsup != 0 && !is_waiting_on_stations ) IssueSystemMessage( "Comms Lost." );
 			commsup = 0;
 			StartWebSocket();
+			setTimeout(doPoll, 1000);
 		}
 		else
 			$('#SystemStatusClicker').prop( "value", "System " + 0 + "Hz" );
@@ -245,7 +243,11 @@ function onMessage(evt)
 		}
 	}
 
+}
+
+function doPoll(){
 	doSend('wx'); //Request RSSI.
+	setTimeout(doPoll, 500);
 }
 
 function onError(evt)
